datumaro/plugins/vgg_face2_format.py

```diff
@@ -22,21 +22,30 @@ class VggFace2Path:
     LABELS_FILE = 'labels.txt'
     IMAGES_DIR_NO_LABEL = 'no_label'
 
-class VggFace2Extractor(SourceExtractor):
-    def __init__(self, path, subset=None):
-        if not osp.isfile(path):
-            raise Exception("Can't read .csv annotation file '%s'" % path)
+class VggFace2Extractor(Extractor):
+    def __init__(self, path):
+        if not osp.isdir(path):
+            raise Exception("Can't read directory with annotations '%s'" % path)
+
+        annotation_files = [p for p in os.listdir(path)
+            if (osp.basename(p).startswith(VggFace2Path.BBOXES_FILE) or \
+                osp.basename(p).startswith(VggFace2Path.LANDMARKS_FILE)) and \
+                p.endswith('csv')]
+
+        if len(annotation_files) < 1:
+            raise Exception("Can't find annotations in the directory '%s'" % path)
+
+        super().__init__()
+
         self._path = path
-        self._dataset_dir = osp.dirname(osp.dirname(path))
+        self._dataset_dir = osp.dirname(path)
 
-        if not subset:
-            subset = osp.splitext(osp.basename(path))[0]
-            if subset.startswith(VggFace2Path.LANDMARKS_FILE):
-                subset = subset.split('_')[2]
-        super().__init__(subset=subset)
+        self._subsets = {osp.splitext(f.split('_')[2])[0] for f in annotation_files}
 
         self._categories = self._load_categories()
-        self._items = list(self._load_items(path).values())
+        self.items = []
+        for subset in self._subsets:
+            self._items.extend(list(self._load_items(subset).values()))
```

Comment:
I suppose that one of this variables is misspelled.

Reply:
Fixed.